rs/nervous_system/integration_tests/src/pocket_ic_helpers.rs

```diff
@@ -2230,28 +2232,34 @@ pub mod sns {
         const NUM_TRANSACTIONS_NEEDED_TO_SPAWN_FIRST_ARCHIVE: u64 = 2000;
 
         // Generate a bunch of SNS token transactions.
-        for i in 0..NUM_TRANSACTIONS_NEEDED_TO_SPAWN_FIRST_ARCHIVE {
-            let user_principal_id = PrincipalId::new_user_test_id(i);
-            let direct_participant_swap_account = Account {
-                owner: user_principal_id.0,
-                subaccount: None,
-            };
-            let _block_height = ledger::icrc1_transfer(
-                pocket_ic,
-                sns_ledger_canister_id,
-                sns_governance_canister_id,
-                TransferArg {
-                    from_subaccount: None,
-                    to: direct_participant_swap_account,
-                    fee: None,
-                    created_at_time: None,
-                    memo: None,
-                    amount: Nat::from(100_000_u64), // mint an arbitrary amount of SNS tokens
-                },
-            )
-            .await
-            .unwrap();
-        }
+        stream::iter(0..NUM_TRANSACTIONS_NEEDED_TO_SPAWN_FIRST_ARCHIVE)
+            .map(|i| {
+                async move {
+                    let user_principal_id = PrincipalId::new_user_test_id(i);
+                    let direct_participant_swap_account = Account {
+                        owner: user_principal_id.0,
+                        subaccount: None,
+                    };
+                    let _block_height = ledger::icrc1_transfer(
+                        pocket_ic,
+                        sns_ledger_canister_id,
+                        sns_governance_canister_id,
+                        TransferArg {
+                            from_subaccount: None,
+                            to: direct_participant_swap_account,
+                            fee: None,
+                            created_at_time: None,
+                            memo: None,
+                            amount: Nat::from(100_000_u64), // mint an arbitrary amount of SNS tokens
+                        },
+                    )
+                    .await
+                    .unwrap();
+                }
+            })
+            .buffer_unordered(100)
```

Comment:
buffer_unordered(100) means up to 100 requests will be sent concurrently. I tried bumping it up to 1000 and it made no difference on my machine compared to 100.